linera-chain/src/data_types.rs

```diff
@@ -766,13 +776,24 @@ impl BlockProposal {
             forced_oracle_responses: Some(executed_block.outcome.oracle_responses),
         };
         let signature = Signature::new(&content, secret);
-        Self {
+        let block_proposal = Self {
             content,
             owner: secret.public().into(),
             signature,
             blobs,
             validated_block_certificate: Some(lite_cert),
-        }
+        };
+        block_proposal.check_size(maximum_block_proposal_size)?;
```

Comment:
That's not strictly necessary: We'd fail already in the local node anyway if it were too large, and this time we serialize it multiple times locally just to check the size. It would also keep the code simpler to remove this.